server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -623,8 +638,17 @@ private void setFlushAfterMergeThresholdSize(ByteSizeValue byteSizeValue) {
         this.flushAfterMergeThresholdSize = byteSizeValue;
     }
 
+    private void setTranslogPruningByRetentionLease(boolean enabled) {
+        this.translogPruningByRetentionLease = enabled;
+        if(enabled) {
+            setTranslogRetentionSize(DEFAULT_TRANSLOG_RETENTION_SIZE);
```

Comment:
Why are we changing retention size based on translog prune setting?

Can't we do something like this?

1. Keep the old tlog retention size logic.
2. Use the size value derived via old tlog retention size logic to determine whether to retain the tlog?

Reply:
Currently, Once the soft deletes are enabled, even if we set translog size settings, It will not get updated. 
Are you suggesting to change this as well?